gems/pending/spec/appliance_console/logical_volume_management_spec.rb

```diff
@@ -0,0 +1,137 @@
+require 'appliance_console/logical_volume_management'
+require 'pathname'
+require 'tmpdir'
+
+describe ApplianceConsole::LogicalVolumeManagement do
+  before do
+    @spec_name = File.basename(__FILE__).split(".rb").first
+    @disk_double = double(@spec_name, :path => "/dev/vtest")
+    @config = described_class.new(:disk => @disk_double, :mount_point => "/mount_point", :name => "test")
+  end
+
+  describe ".new" do
+    it "ensures required disk option is provided" do
+      expect { described_class.new(:mount_point => "/mount_point", :name => "test") }.to raise_error(ArgumentError)
+    end
+
+    it "ensures required mount_point option is provided" do
+      expect { described_class.new(:disk => @disk_double, :name => "test") }.to raise_error(ArgumentError)
+    end
+
+    it "ensures required name option is provided" do
+      expect do
+        described_class.new(:disk => @disk_double, :mount_point => "/mount_point")
+      end.to raise_error(ArgumentError)
+    end
+
+    it "sets derived and default instance variables automatically" do
+      expect(@config.volume_group_name).to eq("vg_test")
+      expect(@config.filesystem_type).to eq("xfs")
+      expect(@config.logical_volume_path).to eq("/dev/vg_test/lv_test")
+    end
+  end
+
+  describe "#setup" do
+    before do
+      expect(@disk_double).to receive(:create_partition_table)
+      expect(@disk_double).to receive(:partitions).and_return([:fake_partition])
+      @config.disk = @disk_double
+
+      @fstab = double(@spec_name)
+      allow(@fstab).to receive_messages(:entries => [])
+      allow(LinuxAdmin::FSTab).to receive_messages(:instance => @fstab)
+
+      expect(AwesomeSpawn).to receive(:run!).with("parted -s /dev/vtest mkpart primary 0% 100%")
+      expect(AwesomeSpawn).to receive(:run!).with("mkfs.#{@config.filesystem_type} /dev/vg_test/lv_test")
+      expect(LinuxAdmin::Disk).to receive(:local).and_return([@disk_double])
+      expect(LinuxAdmin::PhysicalVolume).to receive(:create).and_return(:fake_physical_volume)
+      expect(LinuxAdmin::VolumeGroup).to receive(:create).and_return(:fake_volume_group)
+      expect(FileUtils).to_not receive(:rm_rf).with(@config.mount_point)
+
+      @fake_logical_volume = double(@spec_name, :path => "/dev/vg_test/lv_test")
+      expect(LinuxAdmin::LogicalVolume).to receive(:create).and_return(@fake_logical_volume)
+    end
+
+    context "when mount point is not a symbolic link" do
```

Comment:
Same as the other file, if there's only one spec in a context, just move the before/after into the spec and get rid of the context.